app/notification/template_types.py

```diff
@@ -2,21 +2,25 @@
 
 
 def email_recipient(data: dict, example_data: dict, notification_class):
-    if data.get("type") == "MAGIC_LINK":
-        magic_link = notification_class.send_magic_link(data)
-        return magic_link
-    elif data.get("type") == "NOTIFICATION":
-        return f"Currently {data.get('type')} service is not available."
-    elif data.get("type") == "REMINDER":
-        return f"Currently {data.get('type')} service is not available."
-    elif data.get("type") == "AWARD":
-        return f"Currently {data.get('type')} service is not available."
-    else:
-        raise NotificationError(
-            message=(
-                "Incorrect type, please check the key 'type' & other keys,"
-                f" values from notification data: {data}.\n\nExpected"
-                " type:('MAGIC_LINK' or 'NOTIFICATION' or 'REMINDER' or"
-                f" 'AWARD' )\n\nExample data: {example_data}"
+    match data.get("type"):
+        case "MAGIC_LINK":
+            return notification_class.send_magic_link(data)
+
+        case "NOTIFICATION":
+            return f"Currently {data.get('type')} service is not available."
+
+        case "REMINDER":
+            return f"Currently {data.get('type')} service is not available."
+
+        case "AWARD":
+            return f"Currently {data.get('type')} service is not available."
```

Comment:
You could use an Or pattern here and simplify this https://peps.python.org/pep-0636/#or-patterns

Reply:
Thank you for pointing that out, not sure why didn't I think of it. Updated 😄